client/desktop/lib/notifications/api/notes.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Internal dependencies
+ */
+const state = require( 'desktop/lib/state' );
+const handler = require( 'wpcom-xhr-request' );
+
+const promiseTimeout = function ( ms, promise ) {
+	const timeout = new Promise( ( _, reject ) => {
+		const id = setTimeout( () => {
+			clearTimeout( id );
+			reject( `Request timed out in ${ ms } ms` );
+		}, ms );
+	} );
+
+	return Promise.race( [ promise, timeout ] );
+};
+
+async function fetchNote( noteId ) {
+	return new Promise( ( resolve, reject ) => {
+		handler(
+			{
+				path: `/notifications/${ noteId }`,
+				authToken: state.getUser().token,
+				apiVersion: '1.1',
+				query: {
+					fields: 'id,type,unread,body,subject,timestamp,meta,note_hash',
+				},
+			},
+			( error, body ) => {
+				if ( error ) {
+					reject( error );
+				}
+
+				if ( body.notes && Array.isArray( body.notes ) && body.notes.length > 0 ) {
+					return resolve( body.notes[ 0 ] );
+				}
+
+				return resolve( null );
+			}
+		);
+	} );
+}
+
+async function markReadStatus( noteId, isRead ) {
+	return new Promise( ( resolve, reject ) => {
+		handler(
+			{
+				method: 'POST',
+				path: '/notifications/read',
+				apiVersion: '1.1',
+				authToken: state.getUser().token,
+				body: {
+					counts: {
+						[ noteId ]: isRead ? 9999 : -1,
```

Comment:
Could you add a comment to explain these magic numbers?

Reply:
Added comment in 045aa234d2d30f1f8386bf6c1fe4668b20f3094e